src/components/molecules/ChatList/ChatList.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState } from "react";
+import { User } from "types/User";
+import UserProfileModal from "components/organisms/UserProfileModal";
+import {
+  PrivateChatMessage,
+  RestrictedChatMessage,
+} from "components/context/ChatContext";
+import { useSelector } from "hooks/useSelector";
+import { WithId } from "utils/id";
+import { Modal } from "react-bootstrap";
+import { DEFAULT_PARTY_NAME, DEFAULT_PROFILE_IMAGE } from "settings";
+import { formatUtcSeconds } from "utils/time";
+import { getLinkFromText } from "utils/getLinkFromText";
+import { useUser } from "hooks/useUser";
+import "./ChatList.scss";
+
+interface ChatListProps {
+  messages: WithId<RestrictedChatMessage | PrivateChatMessage>[];
+  emptyListMessage?: string;
+  allowDelete?: boolean;
+  deleteMessage: (id: string) => Promise<void>;
+}
+
+const ChatList: React.FC<ChatListProps> = ({
+  messages,
+  allowDelete,
+  emptyListMessage,
+  deleteMessage,
+}) => {
+  const { user } = useUser();
+  const usersById = useSelector((state) => state.firestore.data.chatUsers);
+  const [selectedUserProfile, setSelectedUserProfile] = useState<
+    WithId<User>
+  >();
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [deleting, setDeleting] = useState(false);
+  const [error, setError] = useState<string | null>();
+  const [messageToDelete, setMessageToDelete] = useState<
+    WithId<RestrictedChatMessage | PrivateChatMessage>
+  >();
+
+  return (
+    <>
+      {!!messages.length && (
+        <div className="chat-messages-container">
+          {usersById &&
+            messages.map(
+              (message: WithId<RestrictedChatMessage | PrivateChatMessage>) => {
+                const sender = { ...usersById[message.from], id: message.from };
+                const profileImageSize = 30;
+                const isMe = sender.id === user!.uid;
+                return (
+                  <React.Fragment
+                    key={`${message.from}-${message.to}-${message.ts_utc}`}
+                  >
+                    {message.from in usersById && (
+                      <>
+                        <div
+                          className={`message chat-message ${
+                            isMe ? "chat-message_own" : ""
+                          }`}
```

Comment:
This dependency looks good, +1 for adding it.

Reply:
Happy for you to do so either here, or the next PR that is useful to you. Otherwise I will likely get to it at some stage when I am doing something more style heavy.